### DISCUSSION TITLE
讓網路科學工具助你成為劇透魔人，資料科學家建立《獵魔士》人際網路圖

### DISCUSSION THREAD

**Discussion by Yao900108** [1101-GS4719-C/Python-Community-GS4719-C#426] [科技新知]
**** 我們常在閱讀完長篇大論的書籍或文章後，雖然大致掌握整個故事的結構，卻無法記得所有人物角色與所有劇情細節。對此，網路科學協助人們以定量與客觀的方式來歸納長篇小說或叢書。
**** 科學家透過撰寫出的電腦程式來篩選、推導並製作出內文句子中每個被提到人物角色的關係度列表，它成了兩個角色是否真的相遇或出現在同個劇情中的最佳指標。在檢查被提到角色之間的關係度之後，定義這個網路中的元素。以單一節點來代表每個角色，並在角色於相同情境或部分內文中被提及時，將有關係的節點相連起來。
[https://technews.tw/2022/03/03/data-scientist-builds-a-detailed-network-map-of-the-witcher/](url)

**Comment by pekoranoinu** [comment]
可是我在閱讀完長篇大論的書籍或文章後，還是無法掌握整個故事的結構，我還有救嗎?

**Comment by yenwei12138** [comment]
以前真的會因為腳色太多太複雜而排斥看小說，有這種程式真的會對劇情理解很有幫助!

**Comment by hapiwangy** [comment]
聽說在NLP(自然語言處裡)裡面就有人在做這類型的事情。聽說校內有人曾經針對笑敖江湖(還是天龍八部，反正是某系列某部有名的小說)的門派等等的進行分析

**Comment by Yuhsin000** [comment]
對我是一個大大的福音ㄟ，每次看過去都忘記

**Comment by Sazmafuji** [comment]
時常看小說看到後面會忘記有些人物，可以透過此回想起來，但我比較感興趣是，透過電腦分析產生的另一種角度的看法，可以透過此增加看小說的樂趣

**Comment by stan890425** [comment]
有時候我自己看小說也繪畫關係圖，可以更清楚脈絡

**Comment by jcpan821** [comment]
這樣會不會反而失去小說原本的韻味阿

**Comment by alexthebigpp** [comment]
真的需要，尤其是過一陣子又跑回來玩可以比較容易進入狀況。

**Comment by Ben6uo** [comment]
謝謝分享 那樣就不用只看別人的解說了

**Comment by etin2468** [comment]
哇，這真的是一個很方便的工具，這樣就不用每次都要重看了

**Comment by Lynn12378** [comment]
對於看過劇情的人可依迅速喚起記憶，感覺超棒的，像我每次回想以前看過的電視劇，都只記得主角@@

**Comment by RogelioKG** [comment]
沒有玩巫師3，當初是因為和朋友打昆特牌，才接觸了獵魔士影集。
第一次看獵魔士的時候是真的需要看整理圖才能搞懂區域地理位置、角色關係和戰略安排。
有了 AI 來整理歸納重點確實對人們來說省下不少時間。
